ftw/upgrade/step.py

```diff
@@ -98,6 +98,21 @@ def objects(self, catalog_query, message, logger=None,
         objects = SavepointIterator.build(objects, savepoints, logger)
         return ProgressLogger(message, objects, logger=logger)
 
+    security.declarePrivate('brains')
+    def brains(self, catalog_query, message, logger=None,
+               savepoints=None):
+        """Queries the catalog (unrestricted) and creates an iterator
+        over the brains.
+        The iterator configures and calls a ``ProgressLogger`` with the
+        passed ``message``.
+        """
+
+        brains = self.catalog_unrestricted_search(
+            catalog_query, full_objects=False)
```

Comment:
you could add some code re-use for the existing `objects` method and this new `brains` method. The only thing that is different is the `full_objects` argument.